pkg/controller/elasticsearch/driver/upgrade_cluster_strategy_test.go

```diff
@@ -0,0 +1,171 @@
+package driver
+
+const (
+	sampleClusterState = `{
```

Comment:
Seems unused at this time, but you said unit tests are still TODO so that is maybe the reason. 
As an aside: I think we should try to avoid relying on cluster state if possible and instead use dedicated APIs like `_cat/shards` or similar. My understanding is that cluster state is purely to be seen as a debugging tool and there are no guarantee wrt API stability

Reply:
Yes, it is intended to be used in some unit tests.
I haven't found deprecation [in the documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-state.html), but if you think that we should use a more appropriate API I'm 👍 

For the moment I would keep it as is because this is the way shards are retrieved in other parts of the code but I can open an issue to discuss it. What do you think ?